aiidalab/app.py

```diff
@@ -393,23 +378,14 @@ def _pip_install(*args, stdout):
             if process.returncode != 0:
                 raise RuntimeError("Failed to install dependencies.")
 
-            # AiiDA plugins require reentry run to be found by AiiDA.
-            if _should_run_reentry():
-                logger.info("\nRunning 'reentry scan'\n")
-                process = run_reentry_scan()
-                for line in io.TextIOWrapper(process.stdout, encoding="utf-8"):
-                    stdout.write(line)
-                process.wait()
-                if process.returncode != 0:
-                    # Let's not fail the install because of this, just emit a warning.
-                    logger.warning("'reentry scan' failed")
-
             # Restarting the AiiDA daemon to import newly installed plugins.
+            # TODO: Skip this if verdi is not available (useful for testing).
             process = run_verdi_daemon_restart()
             for line in io.TextIOWrapper(process.stdout, encoding="utf-8"):
                 stdout.write(line)
             process.wait()
             if process.returncode != 0:
+                # TODO: I don't think we should fail the installation if this step fails.
```

Comment:
It is a bit safe to fail or at least a big warning alert. Because I think restart the daemon here is mainly to make plugins available by daemon. 
It will be a bit confuse for user if they install app but not able to restart daemon, and then they run calculation and get error like "module xxx is not found".

Reply:
Yeah, I am a bit torn on this. The disadvantage here is that one cannot install apps when the daemon is not running, which complicates testing, and also does not make sense conceptually.

In any case this is just a comment, I do not plan to change anything in this PR.

Reply:
> The disadvantage here is that one cannot install apps when the daemon is not running

Maybe just check if it is not running, using `start`? It calls `run_verdi_daemon_restart`, so I guess it has enough flexibility to put a bit logic there?

Reply:
I created #441 to follow up on this.